### ISSUE TITLE
Add data source to get ACM certificate

### ISSUE THREAD

**Issue by tmlye** [hashicorp/terraform-provider-aws#9535]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It is possible to create an ACM certificate via `aws_acm_certificate`, but there is currently no way to retrieve the certificate itself, for example for passing it to a resource that doesn't support using the ARN. I essentially want to get the raw certificate and its certificate chain, as described in the [api documentation](https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html#ACM-GetCertificate-response-Certificate).

### New or Affected Resource(s)

* aws_acm_certificate

### Potential Terraform Configuration

```hcl
data "aws_acm_certificate" "example" {
  domain   = "tf.example.com"
  statuses = ["ISSUED"]
}

resource "unrelated_resource" "unrelated" {
  tls_cert_base64 = "${data.aws_cm_certificate.example.cert}"
}
```

### References

* https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html#ACM-GetCertificate-response-Certificate
* https://github.com/aws/aws-sdk-go/blob/master/service/acm/api.go#L468

**Comment by blckct** [comment]
Curious, what resources don't support ARN?

**Comment by sascha-eglau-dragon** [comment]
> Curious, what resources don't support ARN?

Non-AWS resources, for example a helm chart.

**Comment by tmlye** [comment]
> Curious, what resources don't support ARN?

Non-AWS resources, for example a helm chart.

**Comment by blckct** [comment]
Well, you'd rather have to use [ExportCertificate](https://docs.aws.amazon.com/acm/latest/APIReference/API_ExportCertificate.html) and it works only with private ACM. You can't use normal ACM certificate outside AWS services.

But yeah, sounds like a good idea for private certificates.

**Comment by hectcastro** [comment]
It appears as though a data source [already exists](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/acm_certificate) for `aws_acm_certificate` with the corresponding `statuses` filter.

**Comment by laingsc** [comment]
@hectcastro the data source will return the arn/id of the cert, but not the cert itself.  Looks like this will resolve though. https://github.com/dvdliao/terraform-provider-aws/blob/acmpca-private-cert/aws/data_source_aws_acmpca_private_certificate.go

**Comment by Fuuzetsu** [comment]
>Well, you'd rather have to use ExportCertificate and it works only with private ACM. You can't use normal ACM certificate outside AWS services.

To give another example, you need to give AWS Client VPN an ACM ARN. But you need the public part of this key to put in your OpenVPN config for example to actually make any connections. There's no API to get this directly from the Client VPN endpoint. In AWS console, there's a button to download a config that has this information but there's no API to get it. To make the config yourself (to distribute to users for example), you need the public part. So currently you have to keep the public part around and make sure it's the same as the ACM import you're using. If we could use the `GetCertificate` API, we could just query the ACM ARN and be done with it.

> Looks like this will resolve though. 

ACM PCA does not resolve this: this is for the $400 a month private certificate authority (as per the name). You'll notice you need `certificate_authority_arn` to do anything with this data resource.

Basically we just need something to implement this API call https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html

Or equivalently
```
aws acm get-certificate --certificate-arn …
```

This only gives the public part of the certificate (because to get private parts you have to use ACM PCA) but it is exactly what one may want.
